public/app/plugins/panel/graph/tab_display.html

```diff
@@ -71,12 +71,14 @@ <h5 class="section-heading">Hover tooltip</h5>
 
 		<div class="section gf-form-group">
 			<h5 class="section-heading">Stacking & Null value</h5>
-			<gf-form-switch class="gf-form" label="Stack" label-class="width-7" checked="ctrl.panel.stack" on-change="ctrl.render()">
+			<gf-form-switch class="gf-form" label="Sort as Legend" label-class="width-8" checked="ctrl.panel.sortAsLegend" on-change="ctrl.render()">
```

Comment:
Think a dropdown open named "Stack Order" with two options "By Query" , "By Legend Order". But not sure if that is clearer. Also worth discussing, do we need an option for this? When stacking is this option 95% of the time always desired?

Reply:
Good point!

I bet there might be cases where this is not the case but then people could use series override to get their series to behave exactly as they want.